binaries/s3_binary_upload.py

```diff
@@ -0,0 +1,124 @@
+import argparse
+import glob
+import logging
+import os
+import subprocess
+import sys
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+LOGGER.addHandler(logging.StreamHandler(sys.stderr))
+
+
+REPO_ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")
+
+CONDA_PACKAGES_PATH = os.path.join(REPO_ROOT, "binaries", "conda", "output")
+
+TS_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "dist"))[0]
+MA_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "model-archiver", "dist"))[0]
+WA_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "workflow-archiver", "dist"))[0]
+
+
+class S3BinaryUploader:
+    def __init__(self, s3_bucket: str, s3_backup_bucket: str, is_dryrun: bool = False, is_nightly: bool = False):
```

Comment:
By default, S3 provides backup and guarantees 99.999999999% (11 9's).  I don't think backup bucket is needed for nightly build.